src/libshadertrap/src/executor.cc

```diff
@@ -1221,9 +1252,9 @@ bool Executor::CheckEqualBuffers(CommandAssertEqual* assert_equal) {
   for (size_t index = 0; index < static_cast<size_t>(buffer_size[0]); index++) {
     // We only get here if the calls to glMapBufferRange succeeded, in which
     // case the contents of |mapped_buffer| cannot be null.
-    uint8_t value_1 =
+    T value_1 =
         mapped_buffer[0][index];  // NOLINT(clang-analyzer-core.NullDereference)
-    uint8_t value_2 =
+    T value_2 =
```

Comment:
I think you should change this loop to something like the following:

```
size_t index = 0;
for (auto& format_entry : assert_equal->GetFormatEntries()) {
  switch (format_entry.kind) {
     case kSkip:
         index += format_entry.count;
         break;
     case kByte:
         // Use a loop to compare the buffers byte by byte, for format_entry.count bytes
         // increase index by format_entry.count
     case kFloat:
         // Use a loop to compare the buffers byte by byte, for format_entry.count floats
         // increase index by format_entry.count * sizeof(float)
     etc.
  }
}
```

If you like, you could write a templated helper method to extract out the very similar code that will be required to compare elements of different types.